### PR TITLE
[Active Directory Query v2] support pagination, filter attributes

### PR DESCRIPTION
**Pull request by michal-dagan**
- support pagination (with 3 arguments: page_size, limit, page_cookie), in the following commands: ad-search, ad-get-user, ad-get-computer, ad-get-group-members
- Add to 'ad-get-user' command the `attributes_to_exclude` argument.
- Adjusted the tests to the changes.
- Add test and testPlayBooks.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3599)

## Description
- Added the **limit**, **page-size** and **page-cookie** arguments to the ***ad-get-user***, ***ad-get-computer***, ***ad-get-group-members*** commands.
- Added the **page-size** and **page-cookie** arguments to the ***ad-search*** command.
- Added support for the *attributes-to-exclude* argument in the ***ad-get-user*** command.

## Screenshots
image

image

image

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 1]
```diff
@@ -536,16 +533,25 @@ def free_search(default_base_dn, page_size):
     if attributes:
         attributes = ALL_ATTRIBUTES if attributes == 'ALL' else attributes.split(',')
 
+    page_cookie = args.get('page-cookie')
+    if args.get('page-size'):
+        page_size = arg_to_number(args['page-size'])
+        size_limit = page_size
```
Comment (bziser) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:539]: @yaakovi 
Do we prefer in this case to throw an exception for the user that he entered `page_size` and `size_limit`?

**Review by bziser — COMMENTED** [review]
Great work! :) 
These are my initial comments

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 2]
```diff
@@ -536,16 +533,25 @@ def free_search(default_base_dn, page_size):
...
         time_limit=time_limit,
-        page_size=page_size
+        page_size=page_size,
+        page_cookie=page_cookie
     )
 
-    ec = {} if context_output == 'no' else {'ActiveDirectory.Search(obj.dn == val.dn)': entries['flat']}
+    ec = {} if context_output == 'no' else {'ActiveDirectory.Search(obj.dn == val.dn)': entries['flat'],
+                                            'ActiveDirectory(obj.SearchPageCookie)': {
+                                                'SearchPageCookie': entries['page_cookie']}
```
[the remark this answers is not in the document]
Comment (michal-dagan) [reply Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:553]: returns None

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.yml [hunk 3]
```diff
@@ -142,43 +129,44 @@ script:
       - "no"
       defaultValue: "yes"
       description: Whether to output the search results to the context.
+    - name: page-size
+      description: The page size to query. limit will be ignored.
```
Comment (ShirleyDenkberg) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.yml:133]: There is no limit argument for this command.

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md [hunk 4]
```diff
@@ -155,36 +166,38 @@ ad-create-user username="jack" password="1q2w3e4r!" user-dn="cn=jack,dc=demisto,
...
+| base-dn           | Root. For example, DC=domain,DC=com). By default, the Base DN configured for the instance is used. | Optional | 
+| attributes        | A CSV list of the object attributes to return. For example, "dn,memberOf". To return all object attributes, specify 'ALL'. | Optional | 
+| size-limit        | The maximum number of records to return. Default is 50. | Optional | 
+| time-limit        | The maximum time to pull records (in seconds). | Optional | 
+| context-output    | Whether to output the search results to the context. Possible values are: yes, no. Default is yes. | Optional | 
+| page-size         | The page size to query. The limit value will be ignored. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/README.md:190]: There is no limit argument in this command.

**Comment by ShirleyDenkberg** [comment]
@bziser Doc review completed.

**Review by bziser — APPROVED** [review]

**Pull request merged**